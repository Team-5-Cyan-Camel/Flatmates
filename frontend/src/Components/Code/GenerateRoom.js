import React from "react";
import { useHistory } from "react-router-dom";
import Button from "react-bootstrap/Button";
import axios from "axios";

const GenerateRoom = () => {
  const history = useHistory();
  const MakeRoom = () => {
    // sends request to generate a room
    // redirects to the room with the code
    console.log("generate room");

    axios
      .post("room")
      .then((res) => {
        console.log(res);
        history.push("/room/" + res.data.roomCode);
      })
      .catch(function (error) {
        console.log(error);
      });
  };

  return (
    <>
      <Button onClick={MakeRoom} className="GoButton">
        Generate
      </Button>
    </>
  );
};

export default GenerateRoom;
