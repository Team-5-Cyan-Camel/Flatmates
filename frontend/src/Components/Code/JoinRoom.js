import { useState } from "react";
import { useHistory } from "react-router-dom";

import Button from 'react-bootstrap/Button';


const JoinRoom = () => {
  const history = useHistory();
  let [code, setCode] = useState(undefined);

  const joinRoom = (e) => {
    console.log("Join provided room");
    e.preventDefault();
    // check if code is set
    if (code !== undefined) {
      // need to check if room code exist
      // if does
      history.push("/room/" + code);
    }
  };
  return (
    <>
      <form onSubmit={joinRoom}>
        <h3>Join Room</h3>
        <input class = 'AccountInputField'
          type="text"
          name="code"
          placeholder="Enter Code"
          value={code}
          onChange={(e) => setCode(e.target.value)}
        />
        <br />
      </form>
      <Button

className='GoButton' onClick={joinRoom}
>Join</Button>
    </>
  );
};

export default JoinRoom;
