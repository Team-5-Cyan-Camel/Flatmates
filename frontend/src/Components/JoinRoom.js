import { useState } from "react";
import { useHistory } from "react-router-dom";

const JoinRoom = () => {
  const history = useHistory();
  let [code, setCode] = useState(undefined);

  const joinRoom = (e) => {
    console.log("Join provided room");
    e.preventDefault();
    // need to check if room code exist
    // if does
    history.push("/room/" + code);
  };
  return (
    <>
      <form onSubmit={joinRoom}>
        <h3>or join a room</h3>
        <input
          type="text"
          name="code"
          placeholder="Enter Code here"
          value={code}
          onChange={(e) => setCode(e.target.value)}
        />
        <br />
        <input type="submit" value="Join Room!" />
      </form>
    </>
  );
};

export default JoinRoom;
