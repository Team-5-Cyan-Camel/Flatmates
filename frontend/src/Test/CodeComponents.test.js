import React from "react";
import renderer from "react-test-renderer";
import GenerateRoom from "../Components/Code/GenerateRoom";
import JoinRoom from "../Components/Code/JoinRoom";

test("Room generator button renders correctly", () => {
  const tree = renderer.create(<GenerateRoom />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Join room dialog renders correctly", () => {
  const tree = renderer.create(<JoinRoom />).toJSON();
  expect(tree).toMatchSnapshot();
});
