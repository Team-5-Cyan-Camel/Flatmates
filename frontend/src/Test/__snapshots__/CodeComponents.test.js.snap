// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join room dialog renders correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h3>
      Join Room
    </h3>
    <input
      className="AccountInputField"
      name="code"
      onChange={[Function]}
      placeholder="Enter Code"
      type="text"
      value=""
    />
    <br />
  </form>,
  <button
    className="GoButton btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Join
  </button>,
]
`;

exports[`Room generator button renders correctly 1`] = `
<button
  className="GoButton btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Generate
</button>
`;
